usaspending_api/common/api_request_utils.py

```diff
@@ -309,6 +311,79 @@ def validate(body):
                 raise Exception("Invalid mode, autocomplete modes are 'contains', 'startswith', but got " + body["mode"])
 
 
+class GeoCompleteHandler:
+    """ Handles geographical hierarchy searches """
+
+    def __init__(self, request_body):
+        self.request_body = request_body
+        self.search_fields = {
+          "location_country_code__country_name": {
+            "type": "COUNTRY",
+            "parent": "location_country_code"
+          },
+          "location_state_code": {
+            "type": "STATE",
+            "parent": "location_country_code__country_name"
+          },
+          "location_state_name": {
+            "type": "STATE",
+            "parent": "location_country_code__country_name"
+          },
+          "location_city_name": {
+            "type": "CITY",
+            "parent": "location_county_name"
+          },
+          "location_county_name": {
+            "type": "COUNTY",
+            "parent": "location_state_name"
+          },
+          "location_zip5": {
+            "type": "ZIP",
+            "parent": "location_state_name"
+          },
+          "location_foreign_postal_code": {
+            "type": "POSTAL CODE",
+            "parent": "location_country_code__country_name"
+          },
+          "location_foreign_province": {
+            "type": "PROVINCE",
+            "parent": "location_country_code__country_name"
+          },
+          "location_foreign_city_name": {
+            "type": "CITY",
+            "parent": "location_country_code__country_name"
+          }
+        }
+
+    def build_response(self):
+        # Array of search fields, and their 'parent' fields and types
+        search_fields = self.search_fields
+        value = self.request_body.get("value", None)
+        mode = self.request_body.get("mode", "contains")
+
+        if mode == "contains":
+            mode = "__icontains"
+        elif mode == "startswith":
```

Comment:
If you'll look at line 362 it defaults to "contains" if mode is not specified :)